(For Review) Manual Mod Updating

Bletch1971 opened the pull request:
Part 1 - I have added the functionality to manually update the mod files
based on the ids specified in the ASM settings.

The mod update has been added to the ServerUpdate button and also a new
Mod Update button (near the mod id field).

Bletch1971 committed:
(For Review) Manual Mod Updating

Part 1 - I have added the functionality to manually update the mod files
based on the ids specified in the ASM settings.

The mod update has been added to the ServerUpdate button and also a new
Mod Update button (near the mod id field).

NOTE: A problem exists with the enabling of the new Mod Update button.
It appears to be disable when opening ASM. The button code has been
copied from the Server Update button but I am unable to get it to enable
the same way.

Bletch1971 commented:
Cliff, A problem exists with the enabling of the new Mod Update button.
It is always disabled when opening ASM. The button code has been copied from the Server Update button but I am unable to get it to enable the same way.

ChronosWS commented:
DataContexts are inherited.  It looks like the container for the button (the ScrollViewer on line 694) is setting the DataContext to Profile, so Runtime isn't available there.  You will need to explicitly reference Runtime using another binding path (like DataContext.Runtime, ElementName=SettingsControl).  If you look in the debug window when running the application, you should see a BindingError, which will go away when you have solved this.

Bletch1971 commented:
Thank you so much

On 13 Jan 2016, at 5:37 PM, Cliff Hudson notifications@github.com wrote:

DataContexts are inherited. It looks like the container for the button (the ScrollViewer on line 694) is setting the DataContext to Profile, so Runtime isn't available there. You will need to explicitly reference Runtime using another binding path (like DataContext.Runtime, ElementName=SettingsControl). If you look in the debug window when running the application, you should see a BindingError, which will go away when you have solved this.

—
Reply to this email directly or view it on GitHub.

Bletch1971 committed:
Manual Mod Updating

1. Added option in the setting to include/exclude mods update when
updating server.
2. Fixed Update button enable/disable issue.
3. Now calling external arkmodcopy.exe application instead of code, as
this exe will be used for auto update (eventually).

NOTE: Happy to provide the code to the arkmodcopy program. I will
eventually upload the code to git.

Bletch1971 commented:
OK, I think I have this nailed now, thanks to you.

Manual mod updating has been implemented into ASM. I will start to look at getting the automatic process done too.

Bletch1971 commented:
I you feel this needs to be tested a bit, then I know that a few of the ASM forum members might help out. We have a few regulars, like Beals, Anoobis, Kane.

Bealze commented:
is there  a way i can pull a zip with these changes ill be glade to run it

Bletch1971 commented:
Are you able to compile the code, or do you mean a zip of the executables?

Bletch1971 committed:
Total Conversion

1. Added TotalConversionModId field to the mods section.

Bletch1971 commented:
I am working to implement in the mod update to include SotF (when selected), TotalConversions and also Map mods.

Bletch1971 committed:
Manual Mod Updating - Part 3

1. Have changed the Server Settings Control to include separate mod
sections for Custom map and Total conversions. Each field takes a mod
id.
2. Added a fetch button to the Map Name. Only enabled if Custom map or
TC selected. This button fetches the name of the map from the mod
folder. If the mod is not downloaded then the user is prompted if they
want to download.

Bletch1971 commented:
Checking this in, so I can work on the 232 updates.

Still to do, when starting the server, perform a check that ALL mods have been downloaded, including SotF.

ChronosWS commented on ARK Server Manager/Lib/ModUpdater/ModCopy.cs:
```diff
@@ -0,0 +1,303 @@
+                }
+                return true;
+            }
+            catch
```
Is it possible to get exceptions in the normal course, and should we log something or bubble up an error message?  Unconditional catches are usually an indication of missing failure case handling.

ChronosWS commented on ARK Server Manager/Lib/ModUpdater/ModCopy.cs:
```diff
@@ -0,0 +1,303 @@
+{
+    public static class ModCopy
+    {
+        private class FCompressedChunkInfo
```
Does this code all come from another source?  Are we attributing it?

ChronosWS commented on ARK Server Manager/Lib/ModUpdater/ModUpdater.cs:
```diff
@@ -0,0 +1,134 @@
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"ModUpdater.UpgradeAsync - {ex.Message}");
```
Can we log this to the debug log as well?

ChronosWS commented on ARK Server Manager/Lib/ModUpdater/ModUpdater.cs:
```diff
@@ -0,0 +1,134 @@
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"ModUpdater.GetMapName - {ex.Message}");
```
And here as well.

ChronosWS commented on ARK Server Manager/Lib/Model/ServerRuntime.cs:
```diff
@@ -380,23 +393,117 @@ public static void SendStop(Process proc)
+
+            if (!System.Environment.Is64BitOperatingSystem)
+            {
+                var result = MessageBox.Show("ARK: Survival Evolved(tm) Server requires a 64-bit operating system to run.  Your operating system is 32-bit and therefore the Ark Server Manager will be unable to start the server, but you may still install it or load and save profiles and settings files for use on other machines.\r\n\r\nDo you wish to continue?", "64-bit OS Required", MessageBoxButt ...
```
This needs to come from the localization file.  Were you able to get the run-time query of loc strings from XAML to work?

Bletch1971 committed:
Manual Mod Updating - Part 4

1. Added Map and Mod check when starting the server.

NOTE: No globalization has been done. I would like to do this in a
project specific to implementing globalization.

Bletch1971 commented:
Closing this and re-submitting as it can no longer be merged.
